exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ColumnType.kt

```diff
@@ -985,6 +985,11 @@ class CustomEnumerationColumnType<T : Enum<T>>(
     override fun nonNullValueToString(value: Any): String = super.nonNullValueToString(notNullValueToDB(value))
 }
 
+object UntypedAndUnsizedArrayColumnType : ColumnType() {
+    override fun sqlType(): String =
+        currentDialect.dataTypeProvider.untypedAndUnsizedArrayType()
+}
```

Comment:
I appreciate the need for this and for the scope of this PR it's ok. It does have some problems, for example:
- If a logger is enabled, the array object value is not processed in the output string, which isn't helpful to users wanting to see the values.
- It won't work with edge cases like if a user wants to check if a datetime column value is in an array of LocalDateTimes.

But we need to flesh out the ArrayColumnType more thoroughly anyway, so I can work on these problems after this PR is merged.

Reply:
I didn't remove this because the implemented `Op`s still depend on it. As you commented this will be fleshed out, so to not keep these new definitions in the PR, one way I can think of is to replace its usage with a temporary inlined anonymous `ColumnType` with its `sqlType` being the empty string or just "ARRAY" and the tests still pass.

Reply:
Sure, that sounds like a good temporary option to me.

Reply:
OK, so shall I remove it in the next commit?